dht/range_streamer.cc

```diff
@@ -214,15 +214,15 @@ void range_streamer::add_rx_ranges(const sstring& keyspace_name, std::unordered_
 }
 
 // TODO: This is the legacy range_streamer interface, it is add_rx_ranges which adds rx ranges.
-future<> range_streamer::add_ranges(const sstring& keyspace_name, locator::effective_replication_map_ptr erm, dht::token_range_vector ranges, gms::gossiper& gossiper, bool is_replacing) {
+future<> range_streamer::add_ranges(const sstring& keyspace_name, locator::effective_replication_map_ptr erm, dht::token_range_vector&& ranges, gms::gossiper& gossiper, bool is_replacing) {
```

Comment:
Adding `&&` here doesn't eliminate a copy constructor invocation, but it does eliminate a move constructor invocation (when calling `add_ranges`), which in terms of performance shouldn't really make a difference. It does make the code a bit harder to reason about though, because now the parameter refers to a variable living outside the function. Although it's immediately moved into the lambda so it's still pretty simple.

Reply:
It makes the function exception safe, e.g. if allocating the function that captures `ranges` throws bad_alloc.

Reply:
gone in v7